### DISCUSSION TITLE
Export Feature?

### DISCUSSION THREAD

**Discussion by IngramBenjamin** [mattwright324/youtube-metadata#150] [General]
Hi, I've been using this tool for a few months now and it's been so useful. Thank you. I attempted to use Bulk today, and I'm not seeing the option to export results as a .zip anymore. Has this been removed?

**Comment by mattwright324** [comment]
Hi @IngramBenjamin,

Yes, unfortunately I had to remove it as a result of a YouTube audit ToS violation. It doesn't make sense to me as its not allowing the download of video or playback (what I thought audiovisual content means) but its what they requested as part of the audit.

[Link](https://developers.google.com/youtube/terms/developer-policies#e.-handling-youtube-data-and-content) from picture below

![image](https://github.com/mattwright324/youtube-metadata/assets/7717432/d8371322-116e-4ae8-a74b-f52bb68316bf)

**Reply by mattwright324** [reply]
> Question: I have removed it but I am still confused by the violation of audiovisual content, the stated policy III.E.1.a,b,c (Audiovisual Content) on the website does not mention data in that way. The policy states "download, import, backup, cache, or store copies of YouTube audiovisual content" in section A and "playback" in section B. The exported document zip file contained the json API data and not playback. Could you clarify what audiovisual content means or which channel data is considered as that type of content?

> Answer: As you are allowing users to download data from the website, download. i.e, any kind of YouTube data(Audio, Video, Text files is not allowed to be downloaded .

Still confusing as the violation given to me does not exactly match the [policy on the website](https://developers.google.com/youtube/terms/developer-policies#e.-handling-youtube-data-and-content).

**Reply by parkerjonez** [reply]
@mattwright324 Welcome to the arbitrary and draconian world of YT/Google API policies. Sigh. I remember the good old days when they encouraged development and innovation on the platform. At any rate, sometimes these are just canned responses. Though it looks like a real human responded. 

I thought that the LinkedIn vs. HiQ case would pave the way for this public data to be available. However, it looks like the [outcome wasn't great](https://www2.staffingindustry.com/Editorial/IT-Staffing-Report/Jan.-5-2023/LinkedIn-ends-legal-battle-in-data-scraping-case#:~:text=LinkedIn%20recently%20ended%20its%20six,legal%20win%20against%20hiQ%20Labs.) :( 

Have you appealed there violation since this? I've had luck there in the past.

**Reply by mattwright324** [reply]
Hi @parkerjonez 

Yea it did seem like a real person was responding. Initially I did want to ask about the policy/violation and clarify though a different person to see if I'd get a different response. However, I wasn't sure how or where to do that.

**Reply by parkerjonez** [reply]
I used to work there. Let me see what I can find out. I love what you built, by the way!

**Reply by alimehran200** [reply]
thanks mattw, i found one another similar more easy to use i will suggest , https://searchme.cc/

**Reply by LazytownSpo666** [reply]
> thanks mattw, i found one another similar more easy to use i will suggest , https://searchme.cc/

The link was broken.

**Reply by nurp** [reply]
I really need to look at the description of the last 100 videos of a channel and it stresses my heart to open them all in a separate tab and pause the video, find the description. 
I wonder if you could allow just a small portion to download. I tried to look in the code but there are so many lines, I don't know where they to look.

**Reply by mattwright324** [reply]
Hi @nurp, I assume you are using the Bulk page since you mention multiple videos? You can toggle a bunch of different data columns on and off right above the table, one of those including the description.

![image](https://github.com/user-attachments/assets/55cef674-df77-4443-af1b-9e65bd8238a3)

**Reply by mattwright324** [reply]
If you are using the singular video input page, the description is viewable in the raw json section here.
![image](https://github.com/user-attachments/assets/de37b58d-6693-4818-8ecc-08d8bb2a9129)

**Reply by nurp** [reply]
Thanks, I figured out! :)

**Reply by pogoda-provereno** [reply]
> I really need to look at the description of the last 100 videos of a channel and it stresses my heart to open them all in a separate tab and pause the video, find the description. I wonder if you could allow just a small portion to download. I tried to look in the code but there are so many lines, I don't know where they to look.

Try this one https://github.com/pogoda-provereno/Export-MW-Metadata-Analysis-Results

**Comment by Sebbog1** [comment]
Please add it back again, i would really like to be able to download it all.

**Reply by Waylonaho** [reply]
How can I use this now to get all of the meta data compiled into an easily able way to copy paste into a jsom, csv?

**Comment by shiyw** [comment]
For those who wish the export to come back, try open F12, go to the element tag, and search export.
